[TASK] Document the link builder and how to configure it (#2491)
TYPO3-Documentation/TYPO3CMS-Reference-CoreApi: Documentation/ApiOverview/LinkHandling/LinkBuilder.rst

.. include:: /Includes.rst.txt
.. index:: Link builder
.. _link-builder:

=====================
Frontend link builder
=====================

A link builder, a class extending the abstract class
:php:`\TYPO3\CMS\Frontend\Typolink\AbstractTypolinkBuilder` is called whenever
a link is rendered in the frontend.

There are specific link builders for each type of link. Which link is to

> be called is determined by the class configured in global configuration:

Reviewer: Maybe better: "Which link to call is determined ..."

> There are specific link builders for each type of link. Which link is to
> be called is determined by the class configured in global configuration:

Author: ```suggestion
There are specific link builders for each type of link. Which link to
call is determined by the class configured in global configuration:
```

> :php:`$GLOBALS['TYPO3_CONF_VARS']['FE']['typolinkBuilder'][<type>]`.

Reviewer: Can we use a code snippet and add the additional.php caption to it?

Author: This would be better done in the ext_localconf.php of the extension providing the link builder. Added it.

.. include:: /Includes.rst.txt
.. index:: Link builder
.. _link-builder:

=====================
Frontend link builder
=====================

A link builder, a class extending the abstract class

> :php:`\TYPO3\CMS\Frontend\Typolink\AbstractTypolinkBuilder` is called whenever

Reviewer: ```suggestion
:php:`\TYPO3\CMS\Frontend\Typolink\AbstractTypolinkBuilder`, is called whenever
```

a link is rendered in the frontend.

There are specific link builders for each type of link. Which link to
call is determined by the class configured in global configuration,
see :ref:`typo3ConfVars_fe_typolinkBuilder`.

You can register a custom link builder in your extension's

> :ref:`ext_localconf-php`:

Reviewer: This is not linked.

> :ref:`ext_localconf-php`:

Author: ```suggestion
:ref:`ext-localconf-php`:
```
